src/lib389/lib389/cli_idm/__init__.py

```diff
@@ -111,7 +111,10 @@ def _generic_list(inst, basedn, log, manager_class, args=None):
         if args and args.json:
             json_result = {"type": "list", "items": []}
         for o in ol:
-            o_str = o.__unicode__()
+            # Get the RDN value directly from the DN using ldap.dn.str2dn
+            # This ensures we get the actual RDN value even if keep-old-rdn was used
+            rdn_components = ldap.dn.str2dn(o.dn)[0]
+            o_str = rdn_components[0][1]  # Get the value part of the RDN
```

Comment:
How are rdn with multiple AVAs handled ? (i.e: `cn=John Doe+mail=john.doe@example.com,dc=example,dc=com` ) ?

Reply:
For multiple AVAs handling, we have this ticket opened: https://github.com/389ds/389-ds-base/issues/3329

So yeah, I think it's best to be addressed separately:) I've put the issue number in 'TODO' comment in the code, though.